### ISSUE TITLE
Migrate allowed annotations to Kubernetes recommended syntax

### ISSUE THREAD

**Issue by saschagrunert** [cri-o/cri-o#7781]
CRI-O is not following Kubernetes conventions for annotations. See https://kubernetes.io/docs/reference/labels-annotations-taints/ for what actual Kubernetes annotations look like. This means we could either migrate to the new style or solve the issue for future annotations by enforcing conventions (using docs?).

Refers to: https://github.com/cri-o/cri-o/pull/7779, https://github.com/kubernetes/website/pull/45121#discussion_r1487723768

Migrating the annotations will require a conversion as well as deprecation phase. We did the same some time ago for our metrics, which are still in deprecation:

https://github.com/cri-o/cri-o/blob/7d47844593de87cf6c4fa56c116625f373eae2cd/server/metrics/metrics.go#L70-L83

**Comment by amitamrutiya2210** [comment]
Hey @saschagrunert, I'd like to give this issue a try.

**Comment by amitamrutiya2210** [comment]
Hey @saschagrunert, Can I work on this issue if it's not urgent? As a beginner with cri-o, I may need some days to fix this issue.

**Comment by saschagrunert** [comment]
@amitamrutiya2210, yes, but lets seek consensus with other @cri-o/cri-o-maintainers first before starting to work on this topic.

**Comment by haircommander** [comment]
yeah for this one I'd be fine just adding ones that are within style guidelines, marking the others as deprecated. I don't think we need a config field like we did for the metrics.

**Comment by amitamrutiya2210** [comment]
Hello everyone, I'd like to take a shot at resolving this issue.
/assign

**Comment by kwilczynski** [comment]
Related:

- https://github.com/cri-o/cri-o/pull/7779

There is a review comment there that might be useful.

**Comment by amitamrutiya2210** [comment]
To deprecate the current annotation, we need to declare new variables with updated annotations. I'm seeking suggestions on the variable naming convention for both deprecated and new annotations.

Here's an approach I've considered for naming the variables:

```go
// UsernsMode is the user namespace mode to use
// Deprecated: in favor of UsernsModeAnnotationV2 
UsernsModeAnnotation = "io.kubernetes.cri-o.userns-mode"

// New: UsernsMode is the user namespace mode to use
UsernsModeAnnotationV2 = "userns-mode.kubernetes.crio-o.io"
``` 

This approach is what I've thought of to address this issue. I welcome any suggestions or insights you may have regarding this approach. Your input would be greatly appreciated.

**Note:** Utilizing this approach would require adding a significant amount of code to ensure functionality is available for both the new and deprecated versions.
